### PR TITLE
client, server: add and use runtime dir instead of relying on temp files

### PR DESCRIPTION
**Pull request by haircommander**
This way, we know a conmon-rs's files are cleaned up after its runtime dir is

### CODE REVIEW

**Comment by haircommander** [comment]
maybe a decent TODO: add a test making sure those consts are in sync. linking between rust and go isn't as trivial as go and c, so the approach of conmon didn't seem correct

File: conmon-rs/server/src/config.rs [hunk 1]
```diff
@@ -155,4 +152,14 @@ impl Config {
 
         Ok(())
     }
+    pub fn socket(&self) -> PathBuf {
+        let mut path = self.runtime_dir().clone();
+        path.push(SOCKET);
+        path
```
Comment (saschagrunert) [inline conmon-rs/server/src/config.rs:158]:
```suggestion
        self.runtime_dir().join(SOCKET)
```

File: conmon-rs/server/src/config.rs [hunk 2]
```diff
@@ -155,4 +152,14 @@ impl Config {
...
+    }
+    pub fn conmon_pidfile(&self) -> PathBuf {
+        let mut path = self.runtime_dir().clone();
+        path.push(PIDFILE);
+        path
```
Comment (saschagrunert) [inline conmon-rs/server/src/config.rs:163]:
```suggestion
        self.runtime_dir().join(PIDFILE)
```

File: conmon-rs/server/src/container_io.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 use crate::{streams::Streams, terminal::Terminal};
 use anyhow::{Context, Result};
 use crossbeam_channel::Sender;
+use std::path::Path;
 use std::sync::mpsc::Receiver;
```
Comment (saschagrunert) [inline conmon-rs/server/src/container_io.rs:5]: nit:
```suggestion
use std::{path::Path, sync::mpsc::Receiver};
```

**Comment by haircommander** [comment]
updated as suggested!

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**